.cirrus.yml

```
@@ -0,0 +1,216 @@
+    ARTIFACTORY_DEPLOY_USERNAME: vault-${CIRRUS_REPO_OWNER}-${CIRRUS_REPO_NAME}-qa-deployer
+    ARTIFACTORY_DEPLOY_PASSWORD: VAULT[development/artifactory/token/${CIRRUS_REPO_OWNER}-${CIRRUS_REPO_NAME}-qa-deployer access_token]
+    SIGN_KEY: VAULT[development/kv/data/sign data.key]
+    PGP_PASSPHRASE: VAULT[development/kv/data/sign data.passphrase]
+
+  eks_container:
+    <<: *CONTAINER_DEFINITION
+  maven_cache:
+    folder: ${CIRRUS_WORKING_DIR}/.m2/repository
+  build_script:
+    - source cirrus-env BUILD
+    - regular_mvn_build_deploy_analyze -Dmaven.test.skip=true -Dsonar.skip=true
+  cleanup_before_cache_script: cleanup_maven_repository
+
+
+test_linux_task:
+  # Run unit tests on Linux with Java 11 and SonarQube analysis on Next
+  depends_on:
+    - build
+  <<: *ONLY_IF
+  eks_container:
+    <<: *CONTAINER_DEFINITION
+  env:
+    JDK_VERSION: "11"
+    DEPLOY_PULL_REQUEST: "false"
+  maven_cache:
+    folder: ${CIRRUS_WORKING_DIR}/.m2/repository
+  script:
+    - source cirrus-env QA
+    - PULL_REQUEST_SHA=$GIT_SHA1 regular_mvn_build_deploy_analyze -P-deploy-sonarsource,-release,-sign -Dcommercial -Dmaven.install.skip=true -Dmaven.deploy.skip=true
+  cleanup_before_cache_script: cleanup_maven_repository
+  on_failure:
+    junit_artifacts:
+      path: '**/target/surefire-reports/TEST-*.xml'
+      format: junit
+
+test_windows_task:
+  # Run UTs on Windows (Java 17)
+  depends_on:
+    - build
+  <<: *ONLY_IF
+  ec2_instance:
+    <<: *WINVM_DEFINITION
+  maven_cache:
+    folder: ${CIRRUS_WORKING_DIR}/.m2/repository
+  script:
+    - source cirrus-env QA
+    - source set_maven_build_version $BUILD_NUMBER
+    - mvn -B -e -V verify -Dcommercial -Dmaven.test.redirectTestOutputToFile=false
+  cleanup_before_cache_script: cleanup_maven_repository
+  on_failure:
+    junit_artifacts:
+      path: '**/target/surefire-reports/TEST-*.xml'
+      format: junit
+
+mend_scan_task:
+  # Run Mend analysis
+  depends_on:
+    - build
+  <<: *ONLY_IF_WITH_NIGHTLY
+  <<: *ONLY_MAIN_BRANCHES
+  eks_container:
+    <<: *CONTAINER_DEFINITION
+  env:
+    WS_APIKEY: VAULT[development/kv/data/mend data.apikey]
+    WS_PRODUCTNAME: SonarLint/Core
+    JDK_VERSION: "11"
+  maven_cache:
+    folder: ${CIRRUS_WORKING_DIR}/.m2/repository
+  whitesource_script:
+    - if [[ $CIRRUS_CRON == 'nightly-mend-scan' ]]; then source cirrus-env BUILD-PRIVATE; else source cirrus-env QA; fi
+    - source set_maven_build_version $BUILD_NUMBER
+    - mvn clean install -DskipTests
+    - source ws_scan.sh
+  cleanup_before_cache_script: cleanup_maven_repository
+  allow_failures: "true"
+  always:
+    ws_artifacts:
+      path: "whitesource/**/*"
+
+qa_task:
+  # Run ITs
+  depends_on:
+    - build
+    - test_windows
+  env:
+    ARTIFACTORY_API_KEY: VAULT[development/artifactory/token/${CIRRUS_REPO_OWNER}-${CIRRUS_REPO_NAME}-private-reader access_token]
+    GITHUB_TOKEN: VAULT[development/github/token/licenses-ro token]
```

OOMKilled, try this:
```suggestion
    GITHUB_TOKEN: VAULT[development/github/token/licenses-ro token]
    MAVEN_OPTS: '-Xmx5632m'
```
If it does not work or slows down the build, then you can try to increase the memory a bit more.